resource/schema/list_attribute.go

```diff
@@ -251,7 +251,26 @@ func (a ListAttribute) ValidateImplementation(ctx context.Context, req fwschema.
 		resp.Diagnostics.Append(fwschema.AttributeMissingElementTypeDiag(req.Path))
 	}
 
-	if !a.IsComputed() && a.ListDefaultValue() != nil {
-		resp.Diagnostics.Append(nonComputedAttributeWithDefaultDiag(req.Path))
+	if a.ListDefaultValue() != nil {
+		if !a.IsComputed() {
+			resp.Diagnostics.Append(nonComputedAttributeWithDefaultDiag(req.Path))
+		}
+
+		// Validate Default implementation. This is safe unless the framework
+		// ever allows more dynamic Default implementations at which the
+		// implementation would be required to be validated at runtime.
+		// Reference: https://github.com/hashicorp/terraform-plugin-framework/issues/930
+		defaultReq := defaults.ListRequest{
+			Path: req.Path,
+		}
+		defaultResp := &defaults.ListResponse{}
+
+		a.ListDefaultValue().DefaultList(ctx, defaultReq, defaultResp)
+
+		resp.Diagnostics.Append(defaultResp.Diagnostics...)
+
+		if a.ElementType != nil && !a.ElementType.Equal(defaultResp.PlanValue.ElementType(ctx)) {
+			resp.Diagnostics.Append(fwschema.AttributeDefaultElementTypeMismatchDiag(req.Path, a.ElementType, defaultResp.PlanValue.ElementType(ctx)))
+		}
```

Comment:
Should we still check the equality of the element types if there are diagnostics retrieving the default?

> Same question applies to the other type comparisons in collections/nested attributes

Reply:
I was debating this, but I think you are right. Even though `Default` value handling should be static/offline today, it would probably be annoying to see both diagnostics if, for example, you return an error diagnostic early without setting a default value with a fully realized/valid type. I'll hopefully have some time later today to update this.

Reply:
Does this look okay? https://github.com/hashicorp/terraform-plugin-framework/pull/933/commits/6056a2883a9c829a0e3b235da2cb07b70f6990a6